TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/Tensorflow.java

```diff
@@ -1,49 +1,139 @@
+        TFObjectDetector.Parameters tfodParameters = new TFObjectDetector.Parameters(tfodMonitorId); //creating parameters
+        tfodParameters.minResultConfidence = 0.3f; //minimumConfidenceNecessaryForActingOnDetection
+        tfod = ClassFactory.getInstance().createTFObjectDetector(tfodParameters, vuforia); //create objectDetector
+        tfod.loadModelFromAsset(TFOD_MODEL_ASSET, LABEL_FIRST_ELEMENT, LABEL_SECOND_ELEMENT); //loading models
+        tfod.activate(); //turnOn
+        //Hi, this is Anish, and I'm teaching Vincent and Ethan, how git works.
     }
 
-    public List<Recognition> getInference() {
+    public List<Recognition> getInference() { //get "image" back, a bunch of Recognitions - check out instance variables
         if (tfod != null) {
-            return tfod.getUpdatedRecognitions();
+            return tfod.getUpdatedRecognitions(); //Returns the list of recognitions, but only if they are different than the last call to {@link #getUpdatedRecognitions()}.
         }
         return null;
     }
 
     public void activate() {
-        tfod.activate();
-    }
+        if (tfod != null){
+            tfod.activate();
+        }
+
+    } //activate
 
     public void shutdown() {
-        tfod.shutdown();
+        if (tfod != null){
+            tfod.shutdown();
+        }
+    } //deactivate
+
+    public SquareState getTargetRegion(){
+        if (tfod == null){
+            return null;
+        }
+        List<Recognition> toOperateOffOf = getInference();
+        if (toOperateOffOf.size() == 1){
+            int i = 0;
+            for (Recognition recognitions : toOperateOffOf){
+                telemetry.addData(String.format("label (%d)", i), recognitions.getLabel());
+                telemetry.addData(String.format("  left,top (%d)", i), "%.03f , %.03f",
+                        recognitions.getLeft(), recognitions.getTop());
+                telemetry.addData(String.format("  right,bottom (%d)", i), "%.03f , %.03f",
+                        recognitions.getRight(), recognitions.getBottom());
+                if (recognitions.getConfidence() >= 0.4){
+                    if (recognitions.getLabel().equals("Four")){
+                        telemetry.addData("Returning SquareState", SquareState.BOX_A);
+                        return SquareState.BOX_A;
+                    }
+                    else{
+                        telemetry.addData("Returning SquareState", SquareState.BOX_B);
+                        return SquareState.BOX_B;
+                    }
+                }
+                telemetry.addData("Returning SquareState", SquareState.BOX_C);
+                return SquareState.BOX_C;
+            }
+        }
+        else{
+            telemetry.addData("Returning SquareState", SquareState.BOX_C);
+            return SquareState.BOX_C;
+        }
+        telemetry.addData("Oopsie", null);
+        return null;
+        /*ArrayList<Float> heights = new ArrayList<Float>();
```

Comment:
Don't merge commented-out code into master! This can stay in the branch until it's ready.